module_4/src/route/user.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+import { Router } from 'express';
+import requestValidation, { autoSuggestUsersSchema, userSchema } from '../validators/index.js';
+import { UserController } from '../controllers/index.js';
+import { UserService } from '../service/index.js';
+import { UserQuery } from '../query/index.js';
+
+export const userControllerInst = new UserController(new UserService(new UserQuery()));
+
+export const userRouter = () => {
+
+  const router = Router();
+
+  router.param('id', (req, res, next, id) => userControllerInst.idParamGuard(req, res, next, id));
+
+  router.route('/:id')
+  .get((req, res) => userControllerInst.getUser(req, res))
+  .put(requestValidation(userSchema, 'body'), (req, res) => userControllerInst.updateUser(req, res))
```

Comment:
`validateRequest`
`body` and `query` can be stored in a common enum:
`export const ReqPayloadType = { body: 'body', query: 'query' };`